### ISSUE TITLE
Issue with "Array.prototype.concat()": (short summary here please)

### ISSUE THREAD

**Issue by Sumi0** [mdn/content#6382]
MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat

#### What information was incorrect, unhelpful, or incomplete?
**`valueN Optional`**
`Arrays and/or values to concatenate into a new array. If all valueN parameters are omitted, concat returns a shallow copy of the existing array on which it is called. See the description below for more details.`

#### Specific section or headline?
---**Parameters** Section----

#### What did you expect to see?
---Array.concat() without any parameters passed returns a Deep Copy not a Shallow Copy

#### Did you test this? If so, how?
---On my machine, in browser(Chrome) console.
```
k = [1234, 553, 989, 000, [1,4,5]];
z = k.concat()
z.shift()
console.log('z,k:', z, k)
k.unshift(2)
console.log('z,k:', z, k)
```

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/javascript/reference/global_objects/array/concat`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/javascript/reference/global_objects/array/concat/index.html
* Last commit: https://github.com/mdn/content/commit/13360cbad46b4ce6457f80492d73d2efd252e8a9
* Document last modified: 2021-05-05T10:31:52.000Z

</details>

**Comment by sideshowbarker** [comment]
We can’t change this and shouldn’t — because *shallow copy* is used in the existing article in a way that’s consistent with how that term is commonly used in programming discussions. For example, see the answers at https://stackoverflow.com/questions/184710/what-is-the-difference-between-a-deep-copy-and-a-shallow-copy:

> Shallow copies duplicate as little as possible. A shallow copy of a collection is a copy of the collection structure, not the elements. With a shallow copy, two collections now share the individual elements.
> 
> Deep copies duplicate everything. A deep copy of a collection is two collections with all of the elements in the original collection duplicated.
> 
> In a shallow copy, object B points to object A's location in memory. In deep copy, all things in object A's memory location get copied to object B's memory location.

So “shallow” vs “deep” here is about whether the copy simply just holds shared/borrowed references/shadows of the elements in the source vs whether the copy takes the elements from the source and creates its own standalone duplicates of all those elements.

I guess using the term “shallow” to describe that relationship is misleading — maybe whoever coined it should have instead called it a *“shadow copy”* or “borrow copy” or something. But they named it “shallow copy” instead, and so here we are now.

But anyway, to be clear, despite the name, “shallow copy” doesn’t have anything at all to do with whether, for example, a nested array from the source gets copied. Even in a “shallow” copy, every single element of the source gets reflected exactly as-is the copy — even if some elements in the source are objects which themselves have very deep nesting of other objects within them.

In other words, it’s irrelevant that in `k = [1234, 553, 989, 000, [1,4,5]]` there’s a `[1,4,5]` array nested within that array.

What is relevant is that for `k = [1234, 553, 989, 000, [1,4,5]]; z = k.concat()`, if you do `z[i] === k[i]` for any value of `i`, it’s gonna return true. And that is what tells you that `z` is just a “shallow” copy of `k`.

**Comment by Sumi0** [comment]
Thanks @sideshowbarker , i got it.

Then is this post misleading because, it delivers counter-significance for shallow copy vs deep copy ..?

`In programming, we store values in variables. Making a copy means that you initiate a new variable with the same value(s). However, there is a big potential pitfall to consider: deep copying vs. shallow copying. A deep copy means that all of the values of the new variable are copied and disconnected from the original variable. A shallow copy means that certain (sub-)values are still connected to the original variable.`
https://www.freecodecamp.org/news/copying-stuff-in-javascript-how-to-differentiate-between-deep-and-shallow-copies-b6d8c1ef09cd/

**Comment by sideshowbarker** [comment]
> Thanks @sideshowbarker , i got it.
> 
> Then is this post misleading because, it delivers counter-significance for shallow copy vs deep copy ..?
> 
> `In programming, we store values in variables. Making a copy means that you initiate a new variable with the same value(s). However, there is a big potential pitfall to consider: deep copying vs. shallow copying. A deep copy means that all of the values of the new variable are copied and disconnected from the original variable. A shallow copy means that certain (sub-)values are still connected to the original variable.`
> [freecodecamp.org/news/copying-stuff-in-javascript-how-to-differentiate-between-deep-and-shallow-copies-b6d8c1ef09cd](https://www.freecodecamp.org/news/copying-stuff-in-javascript-how-to-differentiate-between-deep-and-shallow-copies-b6d8c1ef09cd/)

As far as I can see, that is basically saying the same thing that other sources say about what “shallow copy” means.
